src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -306,26 +308,31 @@ private void EntryExpired(CacheEntry entry)
         {
             // TODO: For efficiency consider processing these expirations in batches.
             RemoveEntry(entry);
-            StartScanForExpiredItems();
+            StartScanForExpiredItemsIfNeeded(_options.Clock.UtcNow);
         }
 
         // Called by multiple actions to see how long it's been since we last checked for expired items.
         // If sufficient time has elapsed then a scan is initiated on a background task.
-        private void StartScanForExpiredItems(DateTimeOffset? utcNow = null)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
This isn't inlined otherwise?  Also, ScheduleTask looks pretty small; it's not inlined automatically?

Reply:
before my changes, the `StartScanForExpiredItems` was not getting inlined:

![obraz](https://user-images.githubusercontent.com/6011991/99407260-2ea82b80-28ef-11eb-9163-62ca74b2b7f6.png)

now `StartScanForExpiredItemsIfNeeded` does get inlined an `ScheduleTask` does not:

![obraz](https://user-images.githubusercontent.com/6011991/99407147-10dac680-28ef-11eb-84a6-50874a098ea5.png)

![obraz](https://user-images.githubusercontent.com/6011991/99406924-cbb69480-28ee-11eb-963d-21fbfffb46a4.png)

I was simply not sure if `_options.ExpirationScanFrequency < utcNow - _lastExpirationScan` won't get translated to too much IL and just to be sure I've applied the attribute

Reply:
> I was simply not sure if _options.ExpirationScanFrequency < utcNow - _lastExpirationScan won't get translated to too much IL and just to be sure I've applied the attribute

Ok, but the same argument goes for whether ScheduleTask may get inlined automatically even though you don't want it to.

Reply:
> but the same argument goes for whether ScheduleTask may get inlined automatically even though you don't want it to

It's quite big (from the IL perspective) and I would not expect it to get inlined. Do you want me to ensure it by using an attribute?